modules/lightmapper_rd/lightmapper_rd.cpp

```diff
@@ -458,7 +459,15 @@ void LightmapperRD::_create_acceleration_structures(RenderingDevice *rd, Size2i
 			t.max_bounds[0] = taabb.position.x + MAX(taabb.size.x, 0.0001);
 			t.max_bounds[1] = taabb.position.y + MAX(taabb.size.y, 0.0001);
 			t.max_bounds[2] = taabb.position.z + MAX(taabb.size.z, 0.0001);
-			t.pad0 = t.pad1 = 0; //make valgrind not complain
+
+			t.cull_mode = RendererRD::MaterialStorage::ShaderData::CULL_BACK;
+
+			RID material = mi.data.material[i];
+			if (material.is_valid()) {
+				RendererRD::MaterialStorage::ShaderData *shader_data = material_storage->material_get_shader_data(mi.data.material[i]);
```

Comment:
This crashes when baking from the compatibility renderer. Remember that that the compatibility renderer has its own storage classes. If we need to request this data from the renderer, then it needs to be exposed through the common ancestor MaterialStorage class

Backtrace:
```
================================================================
handle_crash: Program crashed with signal 11
Engine version: Godot Engine v4.3.dev.custom_build (e34b9d1f31eada4e3966f1f4d2ac8bad9b8c6eed)
Dumping the backtrace. Please include this when reporting the bug to the project developer.
[1] /lib/x86_64-linux-gnu/libc.so.6(+0x42520) [0x731fd6e42520] (??:0)
[2] RID_Alloc<RendererRD::MaterialStorage::Material, true>::get_or_null(RID const&, bool) (/usr/bin/../lib/gcc/x86_64-linux-gnu/12/../../../../include/c++/12/bits/atomic_base.h:212)
[3] RID_Owner<RendererRD::MaterialStorage::Material, true>::get_or_null(RID const&) (/home/clayjohn/dev/godot/./core/templates/rid_owner.h:451)
[4] RendererRD::MaterialStorage::get_material(RID) (/home/clayjohn/dev/godot/servers/rendering/renderer_rd/storage_rd/material_storage.h:251)
[5] RendererRD::MaterialStorage::material_get_shader_data(RID) (/home/clayjohn/dev/godot/servers/rendering/renderer_rd/storage_rd/material_storage.cpp:2130)
[6] LightmapperRD::_create_acceleration_structures(RenderingDevice*, Vector2i, int, AABB&, int, unsigned int, Vector<LightmapperRD::Probe>&, Lightmapper::GenerateProbes, Vector<int>&, Vector<int>&, RID&, RID&, RID&, RID&, RID&, RID&, RID&, RID&, RID&, bool (*)(float, String const&, void*, bool), void*) (/home/clayjohn/dev/godot/modules/lightmapper_rd/lightmapper_rd.cpp:467)
[7] LightmapperRD::bake(Lightmapper::BakeQuality, bool, float, int, float, float, int, bool, bool, Lightmapper::GenerateProbes, Ref<Image> const&, Basis const&, bool (*)(float, String const&, void*, bool), void*, float) (/home/clayjohn/dev/godot/modules/lightmapper_rd/lightmapper_rd.cpp:1189)
[8] LightmapGI::bake(Node*, String, bool (*)(float, String const&, void*, bool), void*) (/home/clayjohn/dev/godot/scene/3d/lightmap_gi.cpp:1070)
[9] LightmapGIEditorPlugin::_bake_select_file(String const&) (/home/clayjohn/dev/godot/editor/plugins/lightmap_gi_editor_plugin.cpp:71)
[10] LightmapGIEditorPlugin::_bake() (/home/clayjohn/dev/godot/editor/plugins/lightmap_gi_editor_plugin.cpp:120)
[11] void call_with_variant_args_helper<__UnexistingClass>(__UnexistingClass*, void (__UnexistingClass::*)(), Variant const**, Callable::CallError&, IndexSequence<>) (/home/clayjohn/dev/godot/./core/variant/binder_common.h:309)
[12] void call_with_variant_args_dv<__UnexistingClass>(__UnexistingClass*, void (__UnexistingClass::*)(), Variant const**, int, Callable::CallError&, Vector<Variant> const&) (/home/clayjohn/dev/godot/./core/variant/binder_common.h:452)
[13] MethodBindT<>::call(Object*, Variant const**, int, Callable::CallError&) const (/home/clayjohn/dev/godot/./core/object/method_bind.h:344)
[14] Object::callp(StringName const&, Variant const**, int, Callable::CallError&) (/home/clayjohn/dev/godot/core/object/object.cpp:837)
[15] Callable::callp(Variant const**, int, Variant&, Callable::CallError&) const (/home/clayjohn/dev/godot/core/variant/callable.cpp:69)
[16] Object::emit_signalp(StringName const&, Variant const**, int) (/home/clayjohn/dev/godot/core/object/object.cpp:1201)
[17] Node::emit_signalp(StringName const&, Variant const**, int) (/home/clayjohn/dev/godot/scene/main/node.cpp:3901)
[18] Error Object::emit_signal<>(StringName const&) (/home/clayjohn/dev/godot/./core/object/object.h:933)
[19] BaseButton::_pressed() (/home/clayjohn/dev/godot/scene/gui/base_button.cpp:139)
[20] BaseButton::on_action_event(Ref<InputEvent>) (??:?)
[21] BaseButton::gui_input(Ref<InputEvent> const&) (/home/clayjohn/dev/godot/scene/gui/base_button.cpp:69)
[22] Control::_call_gui_input(Ref<InputEvent> const&) (/home/clayjohn/dev/godot/scene/gui/control.cpp:1796)
[23] Viewport::_gui_call_input(Control*, Ref<InputEvent> const&) (/home/clayjohn/dev/godot/scene/main/viewport.cpp:1538)
[24] Viewport::_gui_input_event(Ref<InputEvent>) (/home/clayjohn/dev/godot/scene/main/viewport.cpp:1805)
[25] Viewport::push_input(Ref<InputEvent> const&, bool) (/home/clayjohn/dev/godot/scene/main/viewport.cpp:3229)
[26] Window::_window_input(Ref<InputEvent> const&) (/home/clayjohn/dev/godot/scene/main/window.cpp:1631)
[27] void call_with_variant_args_helper<Window, Ref<InputEvent> const&, 0ul>(Window*, void (Window::*)(Ref<InputEvent> const&), Variant const**, Callable::CallError&, IndexSequence<0ul>) (/home/clayjohn/dev/godot/./core/variant/binder_common.h:304)
[28] void call_with_variant_args<Window, Ref<InputEvent> const&>(Window*, void (Window::*)(Ref<InputEvent> const&), Variant const**, int, Callable::CallError&) (/home/clayjohn/dev/godot/./core/variant/binder_common.h:419)
[29] CallableCustomMethodPointer<Window, Ref<InputEvent> const&>::call(Variant const**, int, Variant&, Callable::CallError&) const (/home/clayjohn/dev/godot/./core/object/callable_method_pointer.h:104)
[30] Callable::callp(Variant const**, int, Variant&, Callable::CallError&) const (/home/clayjohn/dev/godot/core/variant/callable.cpp:58)
[31] Variant Callable::call<Ref<InputEvent> >(Ref<InputEvent>) const (/home/clayjohn/dev/godot/./core/variant/variant.h:855)
[32] DisplayServerX11::_dispatch_input_event(Ref<InputEvent> const&) (/home/clayjohn/dev/godot/platform/linuxbsd/x11/display_server_x11.cpp:4012)
[33] DisplayServerX11::_dispatch_input_events(Ref<InputEvent> const&) (/home/clayjohn/dev/godot/platform/linuxbsd/x11/display_server_x11.cpp:3989)
[34] Input::_parse_input_event_impl(Ref<InputEvent> const&, bool) (/home/clayjohn/dev/godot/core/input/input.cpp:772)
[35] Input::flush_buffered_events() (/home/clayjohn/dev/godot/core/input/input.cpp:1044)
[36] DisplayServerX11::process_events() (/home/clayjohn/dev/godot/platform/linuxbsd/x11/display_server_x11.cpp:5103)
[37] OS_LinuxBSD::run() (/home/clayjohn/dev/godot/platform/linuxbsd/os_linuxbsd.cpp:958)
[38] /home/clayjohn/dev/godot/bin/godot.linuxbsd.editor.dev.x86_64.llvm(main+0x18e) [0x5863ab6bd93e] (/home/clayjohn/dev/godot/platform/linuxbsd/godot_linuxbsd.cpp:86)
[39] /lib/x86_64-linux-gnu/libc.so.6(+0x29d90) [0x731fd6e29d90] (??:0)
[40] /lib/x86_64-linux-gnu/libc.so.6(__libc_start_main+0x80) [0x731fd6e29e40] (??:0)
[41] /home/clayjohn/dev/godot/bin/godot.linuxbsd.editor.dev.x86_64.llvm(_start+0x25) [0x5863ab6bd6e5] (??:?)
-- END OF BACKTRACE --
================================================================
```

Reply:
Ok so I noticed that VSCode didn't added most of my modifications to git for the Compatibility backend, I'll fix that tomorrow